lib/Map/Vector/ProtomapsGeojsonSource.ts

```diff
@@ -0,0 +1,190 @@
+
+    return geojsonvt(result, {
+      buffer:
+        (PROTOMAPS_TILE_BUFFER / PROTOMAPS_DEFAULT_TILE_SIZE) * geojsonvtExtent,
+      extent: geojsonvtExtent,
+      maxZoom: 24
+    });
+  }
+
+  public async get(
+    c: Zxy,
+    tileSize: number
+  ): Promise<Map<string, ProtomapsFeature[]>> {
+    if (!this.tileIndex) {
+      this.tileIndex = this.fetchData();
+    }
+
+    // request a particular tile
+    const tile = (await this.tileIndex).getTile(c.z, c.x, c.y) as GeojsonVtTile;
+    const result = new Map<string, ProtomapsFeature[]>();
+    const scale = tileSize / geojsonvtExtent;
+
+    if (tile && tile.features && tile.features.length > 0) {
+      result.set(
+        GEOJSON_SOURCE_LAYER_NAME,
+
+        // We have to transform feature objects from GeojsonVtTile to ProtomapsFeature
+        tile.features.map((f) => {
+          let transformedGeom: Point[][] = [];
+          let numVertices = 0;
+
+          // Calculate bbox
+          const bbox: Bbox = {
+            minX: Infinity,
+            minY: Infinity,
+            maxX: -Infinity,
+            maxY: -Infinity
+          };
+          // Multi geometry (eg polygon, multi-line string)
+          if (Array.isArray(f.geometry[0][0])) {
+            const geom = f.geometry as [number, number][][];
+            transformedGeom = geom.map((g1) =>
+              g1.map((g2) => {
+                g2 = [g2[0] * scale, g2[1] * scale];
+                if (bbox.minX > g2[0]) {
+                  bbox.minX = g2[0];
+                }
+
+                if (bbox.maxX < g2[0]) {
+                  bbox.maxX = g2[0];
+                }
+
+                if (bbox.minY > g2[1]) {
+                  bbox.minY = g2[1];
+                }
+
+                if (bbox.maxY < g2[1]) {
+                  bbox.maxY = g2[1];
+                }
+                return new Point(g2[0], g2[1]);
+              })
+            );
+            numVertices = transformedGeom.reduce<number>(
+              (count, current) => count + current.length,
+              0
+            );
+          }
+          // Flat geometry (line string, point)
+          else {
+            const geom = f.geometry as [number, number][];
+            transformedGeom = [
+              geom.map((g1) => {
+                g1 = [g1[0] * scale, g1[1] * scale];
+
+                if (bbox.minX > g1[0]) {
+                  bbox.minX = g1[0];
+                }
+
+                if (bbox.maxX < g1[0]) {
+                  bbox.maxX = g1[0];
+                }
+
+                if (bbox.minY > g1[1]) {
+                  bbox.minY = g1[1];
+                }
+
+                if (bbox.maxY < g1[1]) {
+                  bbox.maxY = g1[1];
+                }
+                return new Point(g1[0], g1[1]);
+              })
+            ];
+            numVertices = transformedGeom.length;
+          }
+
+          const feature: ProtomapsFeature = {
+            props: f.tags,
```

Comment:
The `tags` in f is typed as `any` in this file, but the tag is optional so this will try to assign `something | undefined` to a field of type `JsonObject` (https://github.com/protomaps/protomaps-leaflet/blob/da47e14fe6ba0c4a17c3a8d3dc9921f0f99f39f0/src/tilecache.ts#L13).